src/hooks/useVideoMetadata.ts

```diff
@@ -0,0 +1,42 @@
+import { useEffect, useState } from 'react'
+
+export type VideoMetaData = {
+  type: string
+  sizeInBytes: number
+  height: number
+  width: number
+  duration: number
+}
+export const useVideoMetadata = () => {
+  const [file, setFile] = useState<File>()
+  const [metadata, setMetadata] = useState<VideoMetaData>()
+  useEffect(() => {
+    const videoEl = document.createElement('video')
+    const type = file?.type ?? ''
+    const sizeInBytes = file?.size ?? 0
+    const URL = window.URL || window.webkitURL
```

Comment:
I think we can just safely use `URL` without any fallbacks. It's at 96% support https://caniuse.com/?search=url and we're using other much more cutting edge stuff anyway

Reply:
To clarify, it should be just a global and doesn't need defining